src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp

```diff
@@ -107,24 +107,28 @@ ov::pass::ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
         std::shared_ptr<ov::Node> gather_input_zp = optional_zero_point;
         std::vector<std::shared_ptr<ov::Node>> result_nodes = {};
 
+        // If Convert exists in scale branch, this means that the Scale Constant is stored in non-FP32 precision
+        // The Convert is inserted here to maintain graph correctness. GatherCompressed's output should
+        // follow the Convert's destination precision. The Convert should be kept, and later let the
+        // plugin to decide whether the Convert could be optimized
+        if (pattern_map.count(last_convert_m)) {
+            gather_input_scale = pattern_map.at(last_convert_m).get_node_shared_ptr();
+        }
```

Comment:
I took a look at the transformation, and it seems like it's incorrect to just put `last_convert_m` as `gather_input_scale` because in this case we put **all decompression subgraph** as scale.
But we need to put only scale, but converted to `last_convert_m` precision. To gain that, I suggest cloning `last_convert_m` with new input = `gather_input_scale`. The code should be smth like
```
const auto last_convert = pattern_map.at(last_convert_m).get_node_shared_ptr();
gather_input_scale = last_convert->clone_with_new_inputs({gather_input_scale});
```

Reply:
Thanks a lot for your findings, let me fix it.